### DISCUSSION TITLE
Database Confusion

### DISCUSSION THREAD

**Discussion by gregeeh** [NginxProxyManager/nginx-proxy-manager#2460] [General]
Hi all,

I'm wanting to install NPM in a docker container and I've been reading a lot of setups and some say to setup a mysql or mariadb and some don't even mention a database at all. 

So do I require a separate docker database or not and if not what are the pros and cons of not having one?

TIA

**Comment by Frenzoid** [comment]
Hello gregeeh, by default the npm container has a SQLite database, but if you want to store your data in a different place you can configure an alternative database such as MySQL ( for now it seems that only MySQL is compatible ) in a different container, and pass the credentials to such database to the npm container.

The pros and cons are up to you, but the main difference is having npm store its data on an external database, this might be helpful if you want to re-use a configuration or similar, but as I said, its optional.

**Reply by gregeeh** [reply]
Thank you for the reply, that makes a lot of sense.

**Comment by rj-xy** [comment]
When using an external database, why are the volumes still required?

https://nginxproxymanager.com/setup/#using-mysql-mariadb-database

What is the purpose of the let's encrypt and data volumes if a database is used?
Wouldn't they be redundant?
